redaxo/src/addons/media_manager/lib/media_manager.php

```diff
@@ -17,6 +17,52 @@ public function __construct(rex_managed_media $media)
         $this->useCache(true);
     }
 
+    /**
+     * @param string $type Media type
+     * @param string $file Media file
+     *
+     * @return self
+     */
+    public static function create($type, $file)
+    {
+        $mediaPath = rex_path::media($file);
+        $cachePath = rex_path::addonCache('media_manager');
+
+        $media = new rex_managed_media($mediaPath);
+        $manager = new self($media);
+        $manager->setCachePath($cachePath);
+        $manager->applyEffects($type);
+
+        if ($manager->isCached()) {
+            $media->setSourcePath($manager->getCacheFilename());
+
+            $headers = rex_file::getCache($manager->getHeaderCacheFilename());
+
+            if (isset($headers['__media_manager_format'])) {
```

Comment:
ich finde es unschön dass die daten so vermischt sind. wenn du nen array oder stdclass in das cache file schreibst brauchste den workaround mit dem unset bei den headern auch nicht